src/Elcodi/Component/Geo/View/Address.php

```diff
@@ -0,0 +1,280 @@
+<?php
+
+/*
+ * This file is part of the Elcodi package.
+ *
+ * Copyright (c) 2014-2015 Elcodi.com
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * Feel free to edit as you please, and have fun.
+ *
+ * @author Marc Morera <yuhu@mmoreram.com>
+ * @author Aldo Chiecchia <zimage@tiscali.it>
+ * @author Elcodi Team <tech@elcodi.com>
+ */
+
+namespace Elcodi\Component\Geo\View;
+
+use DateTime;
+
+use Elcodi\Component\Geo\Entity\Interfaces\AddressInterface;
+use Elcodi\Component\Geo\Services\Interfaces\LocationProviderInterface;
+use Elcodi\Component\Geo\ValueObject\LocationData;
+use Elcodi\Component\Geo\View\Interfaces\AddressInterface as AddressViewInterface;
+
+/**
+ * Class Address
+ */
+class Address implements AddressViewInterface
+{
+    /**
+     * @var array
+     *
+     * The hierarchies already retrieved.
+     */
+    private static $hierarchies = [];
```

Comment:
That's static because no matter how's created, this property should be shared. So it only checks the hierarchy once.